esinstall/src/entrypoints.ts

```diff
@@ -0,0 +1,268 @@
+import fs from 'fs';
+import path from 'path';
+import validatePackageName from 'validate-npm-package-name';
+import {ExportField, ExportMapEntry, PackageManifestWithExports, PackageManifest} from './types';
+import {parsePackageImportSpecifier, resolveDependencyManifest} from './util';
+
+// Rarely, a package will ship a broken "browser" package.json entrypoint.
+// Ignore the "browser" entrypoint in those packages.
+const BROKEN_BROWSER_ENTRYPOINT = ['@sheerun/mutationobserver-shim'];
+
+function hasBrokenBrowserEntrypoint(packageName?) {
+  return !!(packageName && BROKEN_BROWSER_ENTRYPOINT.includes(packageName));
+}
+
+function manifestHasTypes(manifest: PackageManifest): boolean {
```

Comment:
nit: `hasTypes` (we try to start these boolean check helper functions with `has`, `is`, etc.